uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/RouteContextSupportTest.java

```diff
@@ -0,0 +1,125 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.eclipse.reddeer.swt.api.TreeItem;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+
+/**
+ * @author fpospisi
+ */
+@RunWith(RedDeerSuite.class)
+public class RouteContextSupportTest {
+	public static final String PROJECT_NAME = "routeContextSupportTest";
+	public static final String ROUTES_NAME = "camel-routes.xml";
+	
+	/**
+	 * <p>
+	 * routeContext support test
+	 * </p>
+	 * <b>Steps</b>
+	 * <ol>
+	 * <li>Create new project: STANDALONE - KARAF - CBR_SPRING.</li>
+	 * <li>In source tab change XML to external <routeContext> "camel-routes.xml" reference.</li>
+	 * <li>Create new XML - camel-routes.xml.</li>
+	 * <li>Edit 'Source' in camel-routes.xml.</li>
+	 * <li>Check camel-routes.xml 'Design' tab for containing items setted up in source. </li>
+	 * <li>Run as Local Camel Context (without tests) and check for Fuse related errors.</li>
+	 * </ol>
+	 */
+	
+	@Before
+	/*
+	 * Create new project.
+	 * */
+	public void setWorkspace() {
+		ProjectFactory.newProject(PROJECT_NAME).deploymentType(STANDALONE).runtimeType(KARAF).template(ProjectTemplate.CBR_SPRING).create();
+	}
+	
+	@After
+	public void cleanWorkspace() {
+		new CleanWorkspaceRequirement().fulfill();
+	}
+	
+	@Test
+	public void testRouteContextSupport() {		
+		/*
+		 * Change content in 'Source'tab for main file.
+		 */
+		CamelEditor.switchTab("Source");
+		EditorManipulator.copyFileContentToCamelXMLEditor("resources/routeContextSupportCamelContext.xml");
+		
+		/*
+		 * Create new XML file 'route-context.xml'.
+		 */
+		create_new_file(ROUTES_NAME);
+
+		/*
+		 * Change content in 'Source' tab for 'route-context.xml'.
+		 */
+		CamelEditor.switchTab("Source");
+		EditorManipulator.copyFileContentToCamelXMLEditor("resources/routeContextSupportRouteContext.xml");				
+		
+		/*
+		 * Check if design tab contains <routeContext>
+		 */
+		check_outline(); 
+		
+		/*
+		 * Run as Local Camel Context (without tests).
+		 */
+		CamelProject project = new CamelProject(PROJECT_NAME);
+		project.openCamelContext("camel-context.xml");
+		CamelEditor.switchTab("Design");		
+		project.runCamelContextWithoutTests("camel-context.xml");
+		assertTrue("Route cbr-route was not property started.", new ConsoleView().getConsoleText().contains("cbr-route started and consuming"));
+		
+		/*
+		 * Check for Fuse related errors.
+		 */
+		LogChecker.assertNoFuseError();		
+	}
+	
+	/*
+	 * Checks if design tab contains <routeContext>
+	 */
+	private void check_outline() {
+		CamelEditor.switchTab("Design");
+		Collection<TreeItem> tree_items = new ContentOutline().outlineElements();
+		String rootOutlineElement = tree_items.stream().findFirst().get().getText();	
+		String rootOutlineElementSplit [] = rootOutlineElement.split(" ", 2);	
+		assertTrue("RouteContext not found", rootOutlineElementSplit[0].equals("RouteContext"));
+	}
+	
+	/*
+	 * Creates new XML file
+	 */
+	private void create_new_file(String name){
```

Comment:
update method name to correct Java method name convention please

Methods | Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized. | run();  runFast();  getBackground();
-- | -- | --